sources/Interop/D3D12MemoryAllocator/include/Allocation.cs

```diff
@@ -0,0 +1,416 @@
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => ref MemoryMarshal.GetReference(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(m_Union.m_Heap), 1));
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct _Anonymous_e__Union
+        {
+            [FieldOffset(0)]
+            public _m_Committed_e__Struct m_Committed;
+
+            [FieldOffset(0)]
+            public _m_Placed_e__Struct m_Placed;
+
+            [FieldOffset(0)]
+            public _m_Heap_e__Struct m_Heap;
+
+            public struct _m_Committed_e__Struct
+            {
+                public D3D12_HEAP_TYPE heapType;
+            }
+
+            public struct _m_Placed_e__Struct
+            {
+                [NativeTypeName("UINT64")]
+                public ulong offset;
+
+                public NormalBlock* block;
+            }
+
+            public struct _m_Heap_e__Struct
+            {
+                public D3D12_HEAP_TYPE heapType;
+                public ID3D12Heap* heap;
+            }
+        }
+
+        internal partial struct PackedData
+        {
+            uint _bitfield0;
+            uint _bitfield1;
+
+            public readonly new Type GetType() => (Type)m_Type;
+
+            public readonly D3D12_RESOURCE_DIMENSION GetResourceDimension() => (D3D12_RESOURCE_DIMENSION)m_ResourceDimension;
+
+            public readonly D3D12_RESOURCE_FLAGS GetResourceFlags() => (D3D12_RESOURCE_FLAGS)m_ResourceFlags;
+
+            public readonly D3D12_TEXTURE_LAYOUT GetTextureLayout() => (D3D12_TEXTURE_LAYOUT)m_TextureLayout;
+
+            [return: NativeTypeName("BOOL")]
+            public readonly int WasZeroInitialized() => (int)m_WasZeroInitialized;
+
+            public void SetType(Type type)
+            {
+                uint u = (uint)type;
+                D3D12MA_ASSERT(u < (1u << 2));
+                m_Type = u;
+            }
+
+            public void SetResourceDimension(D3D12_RESOURCE_DIMENSION resourceDimension)
+            {
+                uint u = (uint)resourceDimension;
+                D3D12MA_ASSERT(u < (1u << 3));
+                m_ResourceDimension = u;
+            }
+
+            public void SetResourceFlags(D3D12_RESOURCE_FLAGS resourceFlags)
+            {
+                uint u = (uint)resourceFlags;
+                D3D12MA_ASSERT(u < (1u << 24));
+                m_ResourceFlags = u;
+            }
+
+            public void SetTextureLayout(D3D12_TEXTURE_LAYOUT textureLayout)
+            {
+                uint u = (uint)textureLayout;
+                D3D12MA_ASSERT(u < (1u << 9));
+                m_TextureLayout = u;
+            }
+
+            public void SetWasZeroInitialized([NativeTypeName("BOOL")] int wasZeroInitialized) => m_WasZeroInitialized = wasZeroInitialized != 0 ? 1 : 0;
+
+            [NativeTypeName("UINT : 2")]
+            private uint m_Type
+            {
+                readonly get => _bitfield0 & 3u;
```

Comment:
nit: ClangSharp generates:
```suggestion
                readonly get => _bitfield0 & 0x3u;
                set => _bitfield0 = (_bitfield0 & ~0x3u) | (value & 0x3u);
```

C# will constant fold the `~0x...` and so it makes it easier to visually verify everything is correct
This would apply to all bitfields in the type.

Reply:
Done in a4fd9deb4b779f81e12b77cea3d63585959fb916, agree it's much clearer now! 😄